control-plane/api-gateway/controllers/gateway_controller.go

```diff
@@ -755,7 +757,7 @@ func (c *GatewayController) getConfigForGatewayClass(ctx context.Context, gatewa
 	if ref := gatewayClassConfig.Spec.ParametersRef; ref != nil {
 		if string(ref.Group) != v1alpha1.GroupVersion.Group ||
 			ref.Kind != v1alpha1.GatewayClassConfigKind ||
-			gatewayClassConfig.Spec.ControllerName != GatewayClassControllerName {
+			string(gatewayClassConfig.Spec.ControllerName) != c.Name {
```

Comment:
since we're already using the `common.GatewayClassControllerName` above, I'd prefer to just stick with that here for minimal changes and consistency rather than passing in the name as a field one place and then using the constant in the `Reconcile` loop above.